test/integration_tests.jl

```diff
@@ -152,3 +140,29 @@
 
     _cleanup_old_branches(url)
 end
+
+@testset "$(service)" for service in [GITHUB, GITLAB]
+    personal_access_token = ENV["INTEGRATION_PAT_$(service)"]
+    test_repo = ENV["INTEGRATION_TEST_REPO_$(service)"]
+
+    env = Dict(
+        "$(service)_REPOSITORY" => test_repo,
+        "$(service)_TOKEN" => personal_access_token,
+        "CI_PROJECT_PATH" => ENV["INTEGRATION_TEST_REPO_GITLAB"]
+    )
+
+    if service == GITHUB
+        env["GITHUB_REPOSITORY"] = "true"
+    elseif service == GITLAB
+        env["GITLAB_CI"] = "true"
+    end
+
+    ci_cfg = CompatHelper.auto_detect_ci_service(env=env)
+    hostname = CompatHelper.api_hostname(ci_cfg)
+
+    api, repo = CompatHelper.get_api_and_repo(ci_cfg, hostname; env=env)
+    user, _ = GitForge.get_user(api)
+    url = CompatHelper.get_url_with_auth(api, hostname, repo)
```

Comment:
`hostname` here needs to be `github.com` or `gitlab.com` instead of the api_hostname that you get above with the `api_hostname` call

Reply:
You can grab the clone hostname with https://github.com/JuliaRegistries/CompatHelper.jl/blob/v3-dev/src/utilities/ci.jl#L27-L28